internal/serialization/types.go

```diff
@@ -1,3 +1,232 @@
 package serialization
 
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/hazelcast/hazelcast-go-client/types"
+
+	"github.com/hazelcast/hazelcast-commandline-client/internal/check"
+)
+
 type NondecodedType string
+
+type JSONValuer interface {
+	JSONValue() (any, error)
+}
+
+type Texter interface {
+	Text() string
+}
+
+func jsonValueToTypeID(v any) int32 {
+	switch v.(type) {
+	case []any:
+		return TypeJavaArray
+	case string:
+		return TypeString
+	case float64:
+		return TypeFloat64
+	case bool:
+		return TypeBool
+	case nil:
+		return TypeNil
+	}
+	return TypeUnknown
+}
+
+type Stringer func(any) string
+
+func staticStringer(v string) Stringer {
+	return func(a any) string {
+		return v
+	}
+}
+
+func sprintStringer(v any) string {
+	if check.IsNil(v) {
+		return ValueNil
+	}
+	return fmt.Sprint(v)
+}
+
+func ptrStringer[T any]() Stringer {
+	return func(v any) string {
+		if vv, ok := v.(T); ok {
+			return fmt.Sprint(vv)
+		}
+		if vv, ok := v.(*T); ok {
+			if vv == nil {
+				return ValueNil
+			}
+			return fmt.Sprint(*vv)
+		}
+		return fmt.Sprintf("?%v?", reflect.TypeOf(v))
+	}
+}
+
+func sprintNilStringer[T any](v *T) string {
+	if v == (*T)(nil) {
+		return ValueNil
+	}
+	return fmt.Sprint(*v)
+}
+
+func arrayStringer[T any](v any) string {
+	vv, ok := v.([]T)
+	if !ok {
+		if vv, ok := v.([]*T); ok {
+			return arrayPtrStringer[T](vv)
+		}
+		return fmt.Sprintf("?%v?", reflect.TypeOf(v))
+	}
+	var sb strings.Builder
+	sb.WriteString("[")
+	if len(vv) > 0 {
+		sb.WriteString(fmt.Sprint(vv[0]))
+		for _, x := range vv[1:] {
+			sb.WriteString(", ")
+			sb.WriteString(fmt.Sprint(x))
+		}
+	}
+	sb.WriteString("]")
+	return sb.String()
+}
+
+func arrayPtrStringer[T any](v any) string {
+	vv := v.([]*T)
+	var sb strings.Builder
+	sb.WriteString("[")
+	if len(vv) > 0 {
+		sb.WriteString(sprintNilStringer(vv[0]))
+		for _, x := range vv[1:] {
+			sb.WriteString(", ")
+			sb.WriteString(sprintNilStringer(x))
+		}
+	}
+	sb.WriteString("]")
+	return sb.String()
+}
+
+func arrayAnySerializer(v any) string {
+	vv, ok := v.([]any)
+	if !ok {
+		return "?array?"
+	}
+	var sb strings.Builder
+	sb.WriteString("[")
+	if len(vv) > 0 {
+		sb.WriteString(sprintStringer(vv[0]))
+		for _, x := range vv[1:] {
+			sb.WriteString(", ")
+			sb.WriteString(sprintStringer(x))
+		}
+	}
+	sb.WriteString("]")
+	return sb.String()
+}
+
+func javaClassStringer(v any) string {
+	return fmt.Sprintf("Java Class: %v", v)
+}
+
+func timeStringer(v any) string {
```

Comment:
Didn't we added this functionality to Go client? Why don't we use it there?

Also instead of creating these stringer functions and registering to a map I think we should bound these methods to types and call `ToString()` directly.